enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/SwitchToSlave.java

```diff
@@ -537,18 +537,58 @@ public void done()
         }
     }
 
+    /**
+     * Monitors events in {@link SwitchToSlave}
+     */
     public interface Monitor
```

Comment:
Nice docs and defaults. Anonymous `SwitchToSlave.Monitor` in `HighAvailabilityMemberStateMachineTest`  can be updated to not override with no-op methods.